lib/tokenizer/preprocessor.js

```diff
@@ -26,15 +31,34 @@ var Preprocessor = module.exports = function () {
 
     this.pos = -1;
     this.lastGapPos = -1;
+    this.lastCharPos = -1;
+    this.droppedBufferSize = 0;
 
     this.gapStack = [];
 
     this.skipNextNewLine = false;
 
     this.lastChunkWritten = false;
     this.endOfChunkHit = false;
+    this.bufferWaterline = DEFAULT_BUFFER_WATERLINE;
 };
 
+Object.defineProperty(Preprocessor.prototype, 'sourcePos', {
+    get: function () {
+        return this.droppedBufferSize + this.pos;
+    }
+});
+
+Preprocessor.prototype.dropParsedChunk = function () {
+    if (this.pos > this.bufferWaterline) {
```

Comment:
Maybe you're right... Just trying to think of the cases where we will still have congestion inside of the streaming parser when chunks were already fed to it, but it didn't output the text node (because it was all text).

Is `bufferWaterline` supposed to be configurable via options btw?